### PR TITLE
Mutect2 WDL and GetSampleName can handle multiple sample names in BAM headers

### PR DESCRIPTION
**Pull request by davidbenjamin**
@rickymagner we need this for Infogain.

### CODE REVIEW

File: scripts/mutect2_wdl/mutect2.wdl [hunk 1]
```diff
@@ -442,22 +442,20 @@ task M2 {
...
-        tumor_command_line="-I ~{tumor_reads} -tumor `cat tumor_name.txt`"
 
         if [[ ! -z "~{normal_reads}" ]]; then
-            gatk --java-options "-Xmx~{command_mem}m" GetSampleName -R ~{ref_fasta} -I ~{normal_reads} -O normal_name.txt -encode \
+            gatk --java-options "-Xmx~{command_mem}m" GetSampleName -R ~{ref_fasta} -I ~{normal_reads} -O normal_names.txt -encode \
             ~{"--gcs-project-for-requester-pays " + gcs_project_for_requester_pays}
-            normal_command_line="-I ~{normal_reads} -normal `cat normal_name.txt`"
+            # add "-normal " to the start of each line and " " to the end, then remove newlines
+            # to get -normal sample1 -normal sample2 etc
+            normal_sample_line=`sed 's/^/-normal /' normal_names.txt | sed 's/$/ /' | tr -d "\n"`
```
Comment (rickymagner) [inline scripts/mutect2_wdl/mutect2.wdl:451]: I think this should be equivalent and slightly less cryptic (unless you hate `awk`):
```
awk '{ print "-normal", $0 }' normal_names.txt | tr '\n' ' '
```

**Review by rickymagner — COMMENTED** [review]
Just a minor comment but looks good otherwise. How many other tools depend on this one, and would any of them possibly break with a multi-sample bam input?

File: scripts/mutect2_wdl/mutect2.wdl (hunk 1 above)
Comment (davidbenjamin) [reply scripts/mutect2_wdl/mutect2.wdl]: done.

**Comment by davidbenjamin** [comment]
The change doesn't affect anything when the BAM has a single sample, and when the BAM has more than one it handles what was previously a failure.  So it's not going to break anything other than a pipeline that was somehow relying on the failure.

**Comment by rickymagner** [comment]
That's a good point. Looks good to me, thanks!

**Pull request merged**